GTBitsOfGood/georgiaCORE
# added navigation tree classes

## Build tree-based data structure to handle navigation info

Issue Number(s): #2 . 

What does this PR change and why?
- Added 3 classes, a tree class, and a normal node, and terminal node class to represent the navigation structure internally.
- Small test file to visually confirm working

### Checklist


### Critical Changes
- Added "type": "module", to `package.json` because it wasn't letting me import my classes

### Related PRs

- #3 

### How to Test

1. node src/navigation/tests/createTreeTest.js
Printed output should match input given in test file.

I didn't do the input format where they give the ID of the next question because I'm not sure this is the best user experience. Regardless I think we should discuss this together and align on a definitive UI because that will inform the final structure of these internal classes. When looking at Jay's progress at the weekly meeting, it seemed like he had a more drag and drop type structure which makes more UX sense. Now we need to figure out if 1. that's the UX we're going with. 2. how we should change my classes to make development as easy as possible.
Some concerns I had are:
- How we're storing this tree in the database, if we need a specific format, if we can have the nested structure I'm currently using, or if we have to define the tree in terms of edges between nodes only, which makes building the tree more difficult (think finding the correct ordering of nodes to connect).
- Does the server/ folder need access to these classes too? In that case we might need to move navigation/ outside of src/
